app/controllers/passwordless/sessions_controller.rb

```diff
@@ -48,6 +48,9 @@ def show
       BCrypt::Password.create(params[:token])
       sign_in passwordless_session
 
+      current_user.role = current_user.roles.first
+      current_user.save
```

Comment:
Hm, not sure why you need to change anything in this controller, let alone persist it?

Reply:
The user needs two forms of role:
1) A collection of roles they can adopt
2) A current role that they are viewing the system in

2 has to be persisted somewhere.  Storing it in the database was a simpler alternative to putting it in the session

Reply:
The purpose of this bit, is when the user logs in, to find them a valid role they can be in (if they haven't picked one yet).
This part was kind of a temporary fix

Reply:
This currently breaks by default on my machine.  I think that there will need to be a database migration to give people roles initially, or just default to a minimum role if one isn't assigned.

Reply:
How does this break?  Is the exception thrown or does it break after that?

Reply:
Rails doesn't like data migrations, could maybe add a rake task to do this?  I wonder if a one off script would be better though

Reply:
Sorry - It might have just been the first time in. It's working now - looking for the screenshot.  I'm wondering if it should throw an exception here?  like Tim said the passwordless session controller should probably be as slim and minimal as possible - it doesn't need to know about roles.  The other controllers have the policies and authorize calls - it feels like any role behaviour belongs there.

Reply:
Looking at removing this code - it will raise an exception later when a user has a null role, are we ok with that as an alternative?

Reply:
@eddieleeper-nb I have an alternative way of doing this now, outside the passwordless controller, better?